test/data/brave-wallet/solana_provider.html

```diff
@@ -20,6 +21,13 @@
     })
   }
 
+  function registerDisconnect() {
+    window.braveSolana.on('disconnect', result => {
+        disconnectEmitted = true
+        window.domAutomationController.send('result ready')
```

Comment:
nit: at some point in future waiting for 'result ready' message should be refactored to waiting for disconnectEmitted promise

Reply:
That is useful when we only want to test the event is emitted but when it is not being emitted there is no obvious way to resolve a different value. If we set a timer to resolve the un-emitted value, the test will be unstable.